User 검색 기능 추가 && Password Hash && 소셜 로그인 기능 추가

cafitac opened the pull request:
### User 검색 기능 추가
UserProfileViewSet에 ListMixin 제거
UserProfileViewSet에 'search' action 추가
### Password Hash 기능 추가
UserSerializer에 (password, password2) 필드 직접 추가 후, 패스워드 validate 추가
UserProfileViewSet에서 'set_password'를 이용하여 password를 hash하는 것으로 변경
### 소셜 로그인 기능 추가
django-rest-auth, django-allauth (install, requirements.txt) 설치
settings 파일에 라이브러리 등록
(Facebook, Kakao, Naver, Google) Login 뷰 추가
(Facebook, KaKao, Naver, Google) 소셜 로그인 url 연결

Commit:
Merge branch 'friend-feature' into member-feature

hanseongd commented on scheduler/member/views.py:
```diff
@@ -11,20 +11,27 @@
-            headers = self.get_success_headers(serializer.data)
+            password = serializer.validated_data['user'].pop('password')
+            serializer.validated_data['user'].pop('password2')
+            user = serializer.save()
```
추후에 user -- > user_profile로 변경하기
둘다 user.user는 직관적이지 않고 나중에 헷갈릴 수 있을 가능성이 있음

hanseongd reviewed (changes requested):
user.user --> userprofile.user

hanseongd commented on scheduler/member/views.py:
```diff
@@ -54,6 +61,29 @@ def update(self, request, *args, **kwargs):
+    def search(self, request):
+        try:
+            user = request.GET.get('name')
+            user = UserProfile.objects.get(user__nickname=user)
```
이곳도

Commit:
User 검색 기능 && Password Hash 기능 && 소셜 로그인

* User 검색 기능
MOD: UserProfileViewSet ListMixin 제거
ADD: UserProfileViewSet에 'search' action 추가

* Password Hash 기능
MOD: UserSerializer에 (password, password2) 필드 직접 추가 후, 패스워드 validate 추가
MOD: UserProfileViewSet에서 'set_password'를 이용하여 password를 hash하는 것으로 변경

* 소셜 로그인
ADD: django-rest-auth, django-allauth (install, requirements.txt) 설치
ADD: settings 파일에 라이브러리 등록
ADD: (Facebook, Kakao, Naver, Google) Login 뷰 추가
ADD: (Facebook, KaKao, Naver, Google) 소셜 로그인 url 연결
